### DISCUSSION TITLE
upgrading Monocular 3D demo on camera not on png file

### DISCUSSION THREAD

**Discussion by damavand1** [open-mmlab/mmdetection3d#2959] [General]
Hi 

I run this demo:
[https://mmdetection3d.readthedocs.io/en/latest/user_guides/inference.html#monocular-3d-demo](https://mmdetection3d.readthedocs.io/en/latest/user_guides/inference.html#monocular-3d-demo)

This demo has 2 limitation
first:
it runs only on 000008.png file 
even if we change file name in source, it don't runs and raise errors, 000008.png file name used in different places
now I want change the source  to runs on the fly and read every frame of camera instead of PNG file on disk

second:
size of 000008.png is 1242*375, and if we pass image with different size it makes some error, I wants this demo runs on every size of image 

this is my repo of base:
[https://github.com/damavand1/mmdetection3d_MySample1](https://github.com/damavand1/mmdetection3d_MySample1)

TNX

**Comment by guizilaile23** [comment]
i have try to do this for 2 days, and for now, i found that the mmdet3D was designed for test local image file only.
the data loader part is too hard to modify.  

openmmlab really makes me feel desperate.

**Reply by damavand1** [reply]
Dear @guizilaile23 

we really needs to get support from openmmlab
or maybe switch to another repository 

Do you know any other opensource repo for 3D object detection and pos estimation with better support ?

I have many unanswered questions even in training
I want to add many objects more than cars and pedestrians like trash bin n  traffic light   and others


also
In this repo they connect to 6 video I think
[ViP3D](https://github.com/Tsinghua-MARS-Lab/ViP3D)
but i don't know how they do that ?

Dear @guizilaile23 if you working in 3D object detection using AI, maybe it's a good idea to share knowledge together, if you agree put a connection way like email or linkedin profile address to connect

tnx

**Reply by guizilaile23** [reply]
heulcl@163.com
